---
title: "Create a NG-CHM"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Create a NG-CHM}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

This vignette demonstrates how to construct a simple NG-CHM, export it to a file, and embed it into RMarkdown.

## Sample NG-CHM Data

This vignette uses an additional package of demo data, [NGCHMDemoData](https://github.com/MD-Anderson-Bioinformatics/NGCHMDemoData), 
which can be installed from [our R-Universe repository](https://md-anderson-bioinformatics.r-universe.dev/packages):

```{r, results='hide', message=FALSE, warning=FALSE, eval=FALSE}
install.packages('NGCHMDemoData', repos = c('https://md-anderson-bioinformatics.r-universe.dev', 'https://cloud.r-project.org'))
```

and loaded into the current environment:

```{r, results='hide', message=FALSE, warning=FALSE }
library(NGCHMDemoData)
```

The sample data includes a matrix of gene expression data, TCGA.BRCA.Expression, containing 3437 genes (rows) and 200 samples 
(columns) of breast cancer data from The Cancer Genome Atlas (TCGA). **IMPORTANT: In order to be used as the basis for an NG-CHM, a matrix should have both
rownames and colnames.** Here the rownames are genes and the colnames are TCGA sample identifiers:

**BRCA Expression Data Matrix:**

```{r}
TCGA.BRCA.ExpressionData[1:4,1:2]
```

The sample data also includes a vector of TP35 mutation status for the TCGA samples in the matrix. This 
data will be used to construct a covariate bar in 
[Covariate Bars and Discrete Color Maps](covariate-bars-and-discrete-color-maps.html). 
**IMPORTANT: In order to be used as the basis for a covariate bar,
a vector should have at least one name in common with the colnames of the matrix.**

**TP53 Mutation Data Vector:**

```{r}
TCGA.BRCA.TP53MutationData[1:2]
```

## Creating a NG-CHM

Using the data loaded above, the `chmNew()` function can be used to create an NG-CHM:

```{r, results='hide', message=FALSE, warning=FALSE }
library(NGCHM)
hm <- chmNew('tcga-brca', TCGA.BRCA.ExpressionData)
```

A covariate bar of the TP53 mutation status can be added with:

```{r, results='hide', message=FALSE, warning=FALSE}
covariateBar <- chmNewCovariate('TP53 Mutation',TCGA.BRCA.TP53MutationData)
hm <- chmAddCovariateBar(hm, 'column', covariateBar)
```


## Export to File

The NG-CHM can be exported to two different file types:

1. A stand-alone HTML file that can be emailed to collaborators, opened in a standard browser, or embedded  
   in an RMarkdown file.
2. A '.ngchm' file that can be opened in the [NGCHM Viewer](https://www.ngchm.net/Downloads/current/ngChmApp.html).

Both methods use files from the [NGCHMSupportFiles](https://github.com/MD-Anderson-Bioinformatics/NGCHMSupportFiles) package 
referenced in the [Installation page](Installation.html). When loaded, NGCHMSupportFiles 
sets environment variables pointing to these additional files.

```{r}
library(NGCHMSupportFiles)
```

The NG-CHM can be exported as a stand-alone HTML file with the `chmExportToHTML()` function.
The first argument is the NG-CHM created above. The second argument is the desired filename.

```{r, results='hide', message=FALSE, warning=FALSE }
chmExportToHTML(hm,'tcga-brca.html')
```

The file 'tcga-brca.html' can be shared with collaborators and opened in a standard web browser.

Alternatively, .ngchm file can be created with the `chmExportToFile()` function.


```{r, eval=FALSE}
chmExportToFile(hm,'tcga-brca.ngchm')
```

The file 'tcga-brca.ngchm' can be opened in the [NG-CHM Viewer](https://www.ngchm.net/Downloads/current/ngChmApp.html).
**IMPORTANT: The filename must end with '.ngchm' to open in the NG-CHM Viewer**. 


## Embed into RMarkdown

The .html file can be embedded into RMarkdown via `htmltools::tags$iframe()`:

```{r}
library('htmltools')
filePath = paste(getwd(),'/tcga-brca.html',sep='')
htmltools::tags$iframe(src=filePath, width='100%', height=700)
```


