---
title: "Getting Started"
output: rmarkdown::html_vignette
resource_files:
  - "tcga-brca.html" # generated by code block at end of vignette and displayed in an iframe
vignette: >
  %\VignetteIndexEntry{Getting Started}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

This vignette demonstrates how to:

- [Format Data for NG-CHMs](#format-data-for-ng-chms)
- [Create an NG-CHM](#create-an-ng-chm)
- [Add a covariate bar](#add-a-covariate-bar)
- [Export to HTML, .ngchm, or PDF](#export-to-html--ngchm-or-pdf)
- [Embed into Rmarkdown](#embed-into-rmarkdown)

Complete code block examples are presented first in [R Code Examples](#r-code-examples),
followed by the sections detailing each step. The fully interactive NG-CHM is in
the last section, [Resulting NG-CHM](#resulting-ng-chm).

Packages used in these examples can be installed with:

```{r, eval=FALSE}
install.packages("NGCHMDemoData", repos = c("https://md-anderson-bioinformatics.r-universe.dev"))
install.packages("NGCHMSupportFiles", repos = c("https://md-anderson-bioinformatics.r-universe.dev"))
install.packages("NGCHM")
```

## R Code Examples

Below are two code examples. The first uses the
[NGCHMDemoData](https://md-anderson-bioinformatics.r-universe.dev/NGCHMDemoData) to create an NG-CHM and export
it to HTML and .ngchm files. The second performs the same task with
generic code containing placeholders for user-specific data files, covariate column names, and other variables.

<details><summary>Click to expand/collapse **code example using NGCHMDemoData**</summary>

```{r, eval=FALSE}
library(NGCHMDemoData)
library(NGCHMSupportFiles)
library(NGCHM)
matrix_data_file <- system.file("extdata", "TCGA.BRCA.Expression.csv", package = "NGCHMDemoData")
matrix_data <- as.matrix(read.csv(matrix_data_file, header = TRUE, row.names = 1, check.names = FALSE))
covariate_data_file <- system.file("extdata", "TCGA.BRCA.TP53Mutation.csv", package = "NGCHMDemoData")
covariate_data <- read.csv(covariate_data_file, row.names = 1) # read.csv returns a data.frame
covariate_vector <- covariate_data[["MutationState"]] # create vector
names(covariate_vector) <- rownames(covariate_data) # set the names
hm <- chmNew("TCGA BRCA Expression", matrix_data)
covariateBar <- chmNewCovariate("TP53 Mutation", covariate_vector)
hm <- chmAddCovariateBar(hm, "column", covariateBar)
chmExportToHTML(hm, "tcga-brca.html", overwrite = TRUE)
chmExportToFile(hm, "tcga-brca.ngchm", overwrite = TRUE)
```
[Back to top](#)

</details>

<details><summary>Click to expand/collapse **generic code example** for .csv input files</summary>

```{r, eval=FALSE}
library(NGCHMSupportFiles)
library(NGCHM)
# In the command below, replace "<path to your matrix data file>" with the path to your matrix data file.
matrix_data_file <- "<path to your matrix data file>"
matrix_data <- as.matrix(read.csv(matrix_data_file, header = TRUE, row.names = 1, check.names = FALSE))
# In the command below, replace "<path to your covariate data file>" with the path to your covariate data file.
covariate_data_file <- "<path to your covariate data file>"
covariate_data <- read.csv(covariate_data_file, row.names = 1) # read.csv returns a data.frame
# In the command below, replace "<covariate column name>" with the name of the column in your covariate data file.
covariate_vector <- covariate_data[["<covariate column name>"]] # create vector
names(covariate_vector) <- rownames(covariate_data) # set the names
# In the command below, replace "<name of your NG-CHM>" with the desired name of your NG-CHM.
hm <- chmNew("<name of your NG-CHM>", matrix_data)
# In the command below, replace "<name of your covariate>" with the desired name of your covariate.
covariateBar <- chmNewCovariate("<name of your covariate>", covariate_vector)
hm <- chmAddCovariateBar(hm, "column", covariateBar)
# In the command below, replace "<name of your HTML file>" with the desired name of your HTML file.
chmExportToHTML(hm, "<name of your HTML file>", overwrite = TRUE)
# In the command below, replace "<name of your .ngchm file>" with the desired name of your .ngchm file.
# IMPORTANT: The filename must end with '.ngchm' in order to open in the NG-CHM Viewer
chmExportToFile(hm, "<name of your .ngchm file>", overwrite = TRUE)
```
[Back to top](#)

</details>

## Format Data for NG-CHMs

For use by the NGCHM package, data must be formatted such that:

- The heat map data is a matrix with rownames and colnames
- The covariate data is a named vector.
  - At least one name in the row/column covariate data vector must be in the row/column names of the heat map matrix.

This example uses the [NGCHMDemoData](https://md-anderson-bioinformatics.r-universe.dev/NGCHMDemoData) package,
which can be installed from [our R-Universe repository](https://md-anderson-bioinformatics.r-universe.dev/packages).

The NGCHMDemoData includes a matrix of gene expression data containing 3437 genes (rows) and 200 samples
(columns) of breast cancer data from The Cancer Genome Atlas (TCGA). This will be converted into a matrix
for constructing the NG-CHM.

<style>
table.expression-data {
  font-family: monospace;
  font-size: 12px;
  width: 90% !important;
  margin-left: auto !important;
  margin-right: auto !important;
}
table.expression-data>tbody>tr>td {
  padding: 5px;
  text-align: right;
}
table.expression-data>thead>tr>th {
  padding: 5px;
  text-align: right;
}
table.mutation-data {
  font-family: monospace;
  font-size: 12px;
  width: 50% !important;
  margin-left: auto !important;
  margin-right: auto !important;
}
table.mutation-data>tbody>tr>td {
  padding: 5px;
  text-align: center;
}
caption {
  text-align: center;
  font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
}
</style>


<table class="expression-data">
  <caption>TCGA.BRCA.Expression.csv snippet</caption>
  <thead>
    <tr class="header">
       <th>    </th>
       <th>TCGA-AO-A0JJ-01A</th>
       <th>TCGA-E9-A1R4-01A</th>
       <th>TCGA-E9-A6HE-01A</th>
       <th>TCGA-E2-A1L9-01A</th>
       <th>\.\.\.</th>
    </tr>
  </thead>
<tbody>
  <tr>
    <td>TSPAN6</td>
    <td>11.838</td>
    <td>9.483</td>
    <td>11.157</td>
    <td>11.536</td>
    <td>\.\.\.</td>
  </tr>
  <tr>
    <td>CFH</td>
    <td>12.033</td>
    <td>10.835</td>
    <td>10.824</td>
    <td>12.878</td>
    <td>\.\.\.</td>
  </tr>
  <tr>
    <td>ENPP4</td>
    <td>11.372</td>
    <td>10.749</td>
    <td>9.721</td>
    <td>11.008</td>
    <td>\.\.\.</td>
  </tr>
  <tr>
    <td>SEMA3F</td>
    <td>12.722</td>
    <td>12.393</td>
    <td>13.338</td>
    <td>12.451</td>
    <td>\.\.\.</td>
  </tr>
  <tr>
    <td>\.\.\.</td>
    <td>\.\.\.</td>
    <td>\.\.\.</td>
    <td>\.\.\.</td>
    <td>\.\.\.</td>
    <td>\.\.\.</td>
  </tr>
</tbody>
</table>


**IMPORTANT: In order to be used as the basis for an NG-CHM, a matrix should have both
rownames and colnames.**

The NGCHMDemoData also includes a vector of TP35 mutation status for the TCGA samples in the matrix. This
data will be used to construct a covariate bar.

<table class="mutation-data">
  <caption>TCGA.BRCA.TP53Mutation.csv snippet</caption>
  <thead>
    <tr class="header">
       <th></th>
       <th>MutationState</th>
    </tr>
  </thead>
<tbody>
  <tr>
    <td>TCGA-AO-A0JJ-01A</td>
    <td>WT</td>
  </tr>
  <tr>
    <td>TCGA-E9-A1R4-01A</td>
    <td>WT</td>
  </tr>
  <tr>
    <td>TCGA-E9-A6HE-01A</td>
    <td>WT</td>
  </tr>
  <tr>
    <td>TCGA-E2-A1L9-01A</td>
    <td>WT</td>
  </tr>
  <tr>
    <td>TCGA-E9-A245-01A</td>
    <td>WT</td>
  </tr>
  <tr>
    <td>TCGA-AO-A0JG-01A</td>
    <td>WT</td>
  </tr>
  <tr>
    <td>TCGA-C8-A1HE-01A</td>
    <td>WT</td>
  </tr>
  <tr>
    <td>TCGA-C8-A12W-01A</td>
    <td>MUT</td>
  </tr>
  <tr>
    <td>\.\.\.</td>
    <td>\.\.\.</td>
  </tr>
</tbody>
</table>

**IMPORTANT: In order to be used as the basis for a row/column covariate bar,
the data should be formatted as a named vector with at least one name in common with the
row/column names of the matrix.**

The NGCHMDemoData package includes these datasets as .csv files. In the code example,
the matrix data is read from the TCGA.BRCA.Expression.csv file (in the NGCHMDemoData package),
and formatted as a matrix. The mutation data is read from the TCGA.BRCA.TP53Mutation.csv file,
and formatted as a named vector.

```{r, eval=FALSE}
library(NGCHMDemoData)
# Read in the NGCHMDemoData expression data, and format as a matrix
matrix_data_file <- system.file("extdata", "TCGA.BRCA.Expression.csv", package = "NGCHMDemoData")
matrix_data <- as.matrix(read.csv(matrix_data_file, header = TRUE, row.names = 1, check.names = FALSE))
# Read in the NGCHMDemoData mutation data, and format as a named vector
covariate_data_file <- system.file("extdata", "TCGA.BRCA.TP53Mutation.csv", package = "NGCHMDemoData")
covariate_data <- read.csv(covariate_data_file, row.names = 1) # read.csv returns a data.frame
covariate_column_name <- "MutationState"
covariate_vector <- covariate_data[[covariate_column_name]] # create vector
names(covariate_vector) <- rownames(covariate_data) # set the names
```

[Back to top](#)

## Create an NG-CHM

The `chmNew()` function creates the NG-CHM object.
The first argument is the name of the NG-CHM, and the second argument is the matrix data.

```{r, eval=FALSE}
library(NGCHM)
hm <- chmNew("TCGA BRCA Expression", matrix_data)
```

[Back to top](#)

## Add a Covariate Bar

The function `chmNewCovariate()` create a covariate bar. The first argument is the name of the covariate,
and the second argument is the named vector of covariate data. The `chmAddCovariateBar()` function adds the
covariate bar to the NG-CHM. The first argument is the NG-CHM, the second argument is the type of covariate bar
('column' or 'row'), and the third argument is the covariate bar object.

```{r, eval=FALSE}
covariateBar <- chmNewCovariate("TP53 Mutation", covariate_vector)
hm <- chmAddCovariateBar(hm, "column", covariateBar)
```

[Back to top](#)

## Export to HTML, .ngchm, or PDF

The NG-CHM can be exported to two different file types:

1. **HTML File**: can be emailed to collaborators, opened in a web browser, or embedded
   in an RMarkdown file.
2. **.ngchm File**: can be opened in the [NGCHM Viewer](https://www.ngchm.net/Downloads/ngChmApp.html).

Both methods use files from the
[NGCHMSupportFiles](https://md-anderson-bioinformatics.r-universe.dev/NGCHMSupportFiles)
package. When loaded, NGCHMSupportFiles sets environment variables pointing to these additional files.

```{r, eval=FALSE}
library(NGCHMSupportFiles)
```

### 1. HTML File

The function `chmExportToHTML()` exports the NG-CHM to an HTML file.
The first argument is the NG-CHM created above. The second argument is the desired filename.
The optional argument `overwrite = TRUE` overwrites the file if it already exists.

```{r, eval=FALSE}
chmExportToHTML(hm, "tcga-brca.html", overwrite = TRUE)
```

The file 'tcga-brca.html' can be shared with collaborators and opened in a web browser.

### 2. .ngchm File

Alternatively, `chmExportToFile()` can be used to create a .ngchm file.


```{r, eval=FALSE}
chmExportToFile(hm, "tcga-brca.ngchm", overwrite = TRUE)
```

The file 'tcga-brca.ngchm' can be opened in the NG-CHM Viewer.
**IMPORTANT: The filename must end with '.ngchm' in order to open in the NG-CHM Viewer**.

### 3. PDF File

To create a publication-quality PDF, there are several options:

<!-- svg symbol symbol taken from NG-CHM viewer:
     https://github.com/MD-Anderson-Bioinformatics/NG-CHM/blob/main/NGCHM/WebContent/icons.svg?short_path=fffa0d9
-->

<svg style="display:none;">
  <symbol id="icon-menu-button-fill" xmlns="http://www.w3.org/2000/svg" fill="currentColor" class="bi bi-menu-button-fill" viewBox="0 0 16 16">
    <path d="M1.5 0A1.5 1.5 0 0 0 0 1.5v2A1.5 1.5 0 0 0 1.5 5h8A1.5 1.5 0 0 0 11 3.5v-2A1.5 1.5 0 0 0 9.5 0h-8zm5.927 2.427A.25.25 0 0 1 7.604 2h.792a.25.25 0 0 1 .177.427l-.396.396a.25.25 0 0 1-.354 0l-.396-.396zM0 8a2 2 0 0 1 2-2h12a2 2 0 0 1 2 2v5a2 2 0 0 1-2 2H2a2 2 0 0 1-2-2V8zm1 3v2a1 1 0 0 0 1 1h12a1 1 0 0 0 1-1v-2H1zm14-1V8a1 1 0 0 0-1-1H2a1 1 0 0 0-1 1v2h14zM2 8.5a.5.5 0 0 1 .5-.5h9a.5.5 0 0 1 0 1h-9a.5.5 0 0 1-.5-.5zm0 4a.5.5 0 0 1 .5-.5h6a.5.5 0 0 1 0 1h-6a.5.5 0 0 1-.5-.5z"/>
  </symbol>
</svg>

1. Create an HTML file [as described above](#html-file). Open the file in a web browser
   and create the view you want to save as a PDF (e.g. the detail map zoomed to the desired location,
   any desired rows/columns selected, covariate bars displayed, etc.). Then click the
<button class="fromViewer">
  <svg style="width:1em;height:1em;vertical-align:middle;" viewBox="0 0 16 16">
    <use xlink:href="#icon-menu-button-fill"></use>
  </svg>
</button>
button (top right corner) and select "Save Heat Map as PDF".

2. Create a .ngchm file [as described above](#ngchm-file).
   Open the .ngchm file in the [NG-CHM Viewer](https://tcga.ngchm.net/NGCHM/chm.html)
   and create the view you want to save as a PDF (e.g. the detail map zoomed to the desired location,
   any desired rows/columns selected, covariate bars displayed, etc.). Then click the
<button class="fromViewer">
  <svg style="width:1em;height:1em;vertical-align:middle;" viewBox="0 0 16 16">
    <use xlink:href="#icon-menu-button-fill"></use>
  </svg>
</button>
button (top right corner) and select "Save Heat Map as PDF".

[Back to top](#)

## Embed into RMarkdown

The HTML file can be embedded into RMarkdown as an iframe using `htmltools::tags`.

```{r, eval=FALSE}
library("htmltools")
htmltools::tags$iframe(src = "tcga-brca.html", width = "100%", height = 700)
```

[Back to top](#)

## Resulting NG-CHM

Below is the fully interactive NG-CHM created in this vignette.

```{r, echo=FALSE, message=FALSE, warning=FALSE}
library(NGCHMDemoData)
library(NGCHMSupportFiles)
library(NGCHM)
library("htmltools")
matrix_data_file <- system.file("extdata", "TCGA.BRCA.Expression.csv", package = "NGCHMDemoData")
matrix_data <- as.matrix(read.csv(matrix_data_file, header = TRUE, row.names = 1, check.names = FALSE))
covariate_data_file <- system.file("extdata", "TCGA.BRCA.TP53Mutation.csv", package = "NGCHMDemoData")
covariate_data <- read.csv(covariate_data_file, row.names = 1) # read.csv returns a data.frame
covariate_column_name <- "MutationState"
covariate_vector <- covariate_data[[covariate_column_name]] # create vector
names(covariate_vector) <- rownames(covariate_data) # set the names
hm <- chmNew("TCGA BRCA Expression", matrix_data)
covariateBar <- chmNewCovariate("TP53 Mutation", covariate_vector)
hm <- chmAddCovariateBar(hm, "column", covariateBar)
invisible(chmExportToHTML(hm, "tcga-brca.html", overwrite = TRUE))
invisible(chmExportToFile(hm, "tcga-brca.ngchm", overwrite = TRUE))
htmltools::tags$iframe(src = "tcga-brca.html", width = "100%", height = 700)
```

[Bact to top](#)

