---
title: "UMAP"
output: rmarkdown::html_vignette
resource_files:
  - "umap.html" # generated by code block at end of vignette and displayed in an iframe
vignette: >
  %\VignetteIndexEntry{UMAP}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
# Load packages here, so startup messages aren't displayed in the vignette
suppressPackageStartupMessages({
  library(NGCHMDemoData)
  library(NGCHM)
})
```

This vignette demonstrates how to perform uniform manifold approximation
and projection (UMAP), add the resulting UMAP coordinates to an
NG-CHM, and explore the interactive features between dimensionality reduction plots and
the NG-CHM via the
[2D Scatter Plot Plugin](https://github.com/MD-Anderson-Bioinformatics/ScatterPlotPlugin).
A similar analysis can be performed for use with the
[3D Scatter Plot Plugin](https://github.com/MD-Anderson-Bioinformatics/ScatterPlotPlugin3D).


## Uniform Mainfold Approximation and Projection (UMAP)

The code block below reads in the
[NGCHMDemoData](https://md-anderson-bioinformatics.r-universe.dev/NGCHMDemoData),
performs principal component analysis (PCA), and calculates UMAP coordinates
from the principal components. This vignette uses the [umap R package](https://cran.r-project.org/package=umap)
which must be installed in order to run the code below.
(See [Getting Started](getting-started.html) for details on creating NG-CHMs).
A static plot of the UMAP coordinates, colored by TP53 mutation state is displayed in Figure 1.

PCA was performed first because it yielded
better group separation compared to performing UMAP on the raw data.

```{r, messsage=FALSE, warning=FALSE}
# Read in NGCHMDemoData (as in the Getting Started vignette)
library(NGCHMDemoData)
matrix_data_file <- system.file("extdata", "TCGA.BRCA.Expression.csv", package = "NGCHMDemoData")
matrix_data <- as.matrix(read.csv(matrix_data_file, header = TRUE, row.names = 1, check.names = FALSE, stringsAsFactors = FALSE))
covariate_data_file <- system.file("extdata", "TCGA.BRCA.TP53Mutation.csv", package = "NGCHMDemoData")
covariate_data <- read.csv(covariate_data_file, row.names = 1, check.names = FALSE, stringsAsFactors = FALSE) # read.csv returns a data.frame
covariate_vector <- covariate_data[["MutationState"]] # create vector
names(covariate_vector) <- rownames(covariate_data) # set the names
# Calculate principal components
pca_data <- prcomp(as.data.frame(t(matrix_data)), scale = TRUE, center = TRUE, rank = 10)
# Calculate UMAP from principal components
library(umap)
config <- umap::umap.defaults
config$n_neighbors <- 15 # change default for better group separation
config$random_state <- 123 # set random state for reproducibility
umap_data <- umap::umap(pca_data$x, config = config)
```

<details><summary>Click to expand R code used to create static plot in Figure 1</summary>

```{r, eval=FALSE}
# Create static plot of UMAP coordinates colored by TP53 mutation state
par(mar = c(4, 4, 4, 8) + 0.1, bg = "white", mgp = c(0.5, 1, 0))
xlim <- range(umap_data$layout[,1])
ylim <- range(umap_data$layout[,2])
plot(xlim, ylim,
     xlab = "UMAP 1",
     ylab = "UMAP 2",
     main = "UMAP",
     type = "n", xaxt = "n", yaxt = "n")
labels_for_point_color <- as.factor(covariate_vector) # "MUT' and "WT"
colors <- c("#f7ef81", "#ffc2e2") # plot only needs two colors: one for "MUT" and one for "WT"
points(umap_data$layout[,1], umap_data$layout[,2],
       col = colors[as.integer(labels_for_point_color)],
       pch = 19, cex = 1.5)
legend(x = xlim[2] + (xlim[2] - xlim[1]) * 0.1, # calculate x position for legend
       y = ylim[2], # calculate y position for legend
       legend = as.character(unique(labels_for_point_color)),
       col = colors[as.integer(unique(labels_for_point_color))],
       title = "TP53 Mutation State",
       inset = 0.03, xpd = TRUE, bty = "n", pch = 19, cex = 0.85)
```
</details>

<!-- If we evaluate the code in the details tag, the plot will be hidden when the
     deails tag is collapsed. So we have two copies of the plotting code: one in the details
     tag to let the user expand/collapse, and the second hidden but evalluated copy here to
     always display the plot.
-->

```{r, echo=FALSE, messsage=FALSE, warning=FALSE, out.width="100%", fig.cap="*Figure 1.* UMAP plot of TCGA BRCA data colored by TP53 mutation state"}
# Create static plot of UMAP coordinates colored by TP53 mutation state
par(mar = c(4, 4, 4, 8) + 0.1, bg = "white", mgp = c(0.5, 1, 0))
xlim <- range(umap_data$layout[,1])
ylim <- range(umap_data$layout[,2])
plot(xlim, ylim,
     xlab = "UMAP 1",
     ylab = "UMAP 2",
     main = "UMAP",
     type = "n", xaxt = "n", yaxt = "n")
labels_for_point_color <- as.factor(covariate_vector) # "MUT' and "WT"
colors <- c("#f7ef81", "#ffc2e2") # plot only needs two colors: one for "MUT" and one for "WT"
points(umap_data$layout[,1], umap_data$layout[,2],
       col = colors[as.integer(labels_for_point_color)],
       pch = 19, cex = 1.5)
legend(x = xlim[2] + (xlim[2] - xlim[1]) * 0.1, # calculate x position for legend
       y = ylim[2], # calculate y position for legend
       legend = as.character(unique(labels_for_point_color)),
       col = colors[as.integer(unique(labels_for_point_color))],
       title = "TP53 Mutation State",
       inset = 0.03, xpd = TRUE, bty = "n", pch = 19, cex = 0.85)
```

[Back to top](#)

## Add UMAP to NG-CHM

This section describes how to add the UMAP coordinates calculated [above](#umap-coordinates)
to the NG-CHM such that they can be explored interactively via the 2D Scatter Plot plugin.

### Create an NG-CHM

This code block creates an NG-CHM from the data read in above and creates a covariate
bar for the TP53 mutation state. The colors of the TP53 mutation state
are chosen to match the colors in the UMAP plot above.

```{r, warning=FALSE, message=FALSE}
library(NGCHM)
hm <- chmNew("TCGA BRCA Expression", matrix_data)
colors <- c("#f7ef81", "#ffc2e2") # same colors as in Figure 1
mutationColorMap <- chmNewColorMap(c("MUT", "WT"), colors)
covariateBar <- chmNewCovariate("TP53 Mutation State", covariate_vector, mutationColorMap)
hm <- chmAddCovariateBar(hm, "column", covariateBar)
```

### Add UMAP Coordinates

The UMAP coordinates are added to the NG-CHM via the convenience
function `chmAddUMAP()`. Similar functions exist for adding PCA, TSNE, etc.
See the "Add Scatter Plot Coordinates" section on the
[Function Reference](../reference/index.html#add-scatter-plot-coordinates) page for more details.
UMAP coordinates can also be added in a fashion similar to the TP53 mutation state.

<!-- Don't evaluate in order to avoid messages from chmExportToHTML (that can seemingly only
     be hidden with `invisible()`. The code block at the end of the vignette adds the umap
     covariate and generates the embedded NGCHM
-->

```{r, warning=FALSE, message=FALSE, eval=FALSE}
hm <- chmAddUMAP(hm, "column", umap_data)
chmExportToHTML(hm, "umap.html", overwrite = TRUE) # create HTML file of NG-CHM
```

## Interactive UMAP / NG-CHM

<!-- svg symbols taken from NG-CHM viewer:
     https://github.com/MD-Anderson-Bioinformatics/NG-CHM/blob/main/NGCHM/WebContent/icons.svg?short_path=fffa0d9
-->

<svg style="display:none;">
  <symbol id="icon-four-panels" xmlns="http://www.w3.org/2000/svg" stroke="currentColor" viewBox="0 0 16 16">
    <rect x="0" y="0" width="8" height="8" fill="#e7fffc"/>
    <rect x="8" y="0" width="8" height="8" fill="#ffe7e8"/>
    <rect x="0" y="8" width="8" height="8" fill="#e7ffe9"/>
    <rect x="8" y="8" width="8" height="8" fill="#fffbe7"/>
  </symbol>
  <symbol id="icon-gear" xmlns="http://www.w3.org/2000/svg" fill="currentColor" class="bi bi-gear-fill" viewBox="0 0 16 16">
    <path d="M9.405 1.05c-.413-1.4-2.397-1.4-2.81 0l-.1.34a1.464 1.464 0 0 1-2.105.872l-.31-.17c-1.283-.698-2.686.705-1.987 1.987l.169.311c.446.82.023 1.841-.872 2.105l-.34.1c-1.4.413-1.4 2.397 0 2.81l.34.1a1.464 1.464 0 0 1 .872 2.105l-.17.31c-.698 1.283.705 2.686 1.987 1.987l.311-.169a1.464 1.464 0 0 1 2.105.872l.1.34c.413 1.4 2.397 1.4 2.81 0l.1-.34a1.464 1.464 0 0 1 2.105-.872l.31.17c1.283.698 2.686-.705 1.987-1.987l-.169-.311a1.464 1.464 0 0 1 .872-2.105l.34-.1c1.4-.413 1.4-2.397 0-2.81l-.34-.1a1.464 1.464 0 0 1-.872-2.105l.17-.31c.698-1.283-.705-2.686-1.987-1.987l-.311.169a1.464 1.464 0 0 1-2.105-.872l-.1-.34zM8 10.93a2.929 2.929 0 1 1 0-5.86 2.929 2.929 0 0 1 0 5.858z"/>
  </symbol>
  <symbol id="icon-big-x" xmlns="http://www.w3.org/2000/svg" fill="currentColor" viewBox="0 0 16 16">
    <path d="m0 0 1 0 7 7 7-7 1 0 0 1-7 7 7 7 0 1 -1 0-7-7-7 7-1 0 0-1 7-7-7-7z"/>
  </symbol>
</svg>

This section describes how to use the
[NG-CHM Viewer](https://bioinformatics.mdanderson.org/public-software/ngchm/) to explore the UMAP data interactively.
NG-CHMs include several plugins, among them a 2D Scatter Plot
to allow for interactive exploration of 2-dimensional data.
Below are the steps to open this plugin and use it to explore the UMAP coordinates.

*It may be helpful to click the "Open NG-CHM in a New Tab" button below
and follow these steps in the larger space of a new tab.*

1. In the "Heat Map Summary" panel, click the
<button class="fromViewer">
  <svg class="fromViewer" viewBox="0 0 16 16">
    <use xlink:href="#icon-four-panels"></use>
  </svg>
</button>
button to open the panel menu.
2. In the newly opened panel menu, under <span class="ngchm-panel-menu-headings">Panel Control</span>,
   select "<span class="ngchm-panel-menu">Add Panel Below</span>".
3. In the newly created empty panel, click the
<button class="fromViewer">
  <svg style="width:16px;height:16px;vertical-align:middle;" viewBox="0 0 16 16">
    <use xlink:href="#icon-four-panels"></use>
  </svg>
</button>
   button, and under <span class="ngchm-panel-menu-headings">Set content to</span>, select
   "<span class="ngchm-panel-menu">2D ScatterPlot</span>". This will automatically open the
<button class="fromViewer">
  <svg class="fromViewer" viewBox="0 0 16 16">
    <use xlink:href="#icon-gear"></use>
  </svg>
</button>
menu for that panel.
4. In the newly opened
<button class="fromViewer">
  <svg class="fromViewer" viewBox="0 0 16 16">
    <use xlink:href="#icon-gear"></use>
  </svg>
</button>
menu, keep the default choices.
5. At the bottom of the open
<button class="fromViewer">
  <svg class="fromViewer" viewBox="0 0 16 16">
    <use xlink:href="#icon-gear"></use>
  </svg>
</button>
menu, click <span class="ngchm-gear-menu-button">APPLY</span> and then <span class="ngchm-gear-menu-button">CLOSE</span>.

The UMAP scatter plot should be visible in the lower right panel of the NG-CHM display.

Here are some suggestions for exploring the interactive features between the UMAP plot and the NG-CHM:

- In the 2D Scatter Plot Panel:
  - Use the slider to increase the size of the points
  - Click the lasso button to enter Lasso/Select mode
  - Draw a lasso around a cluster of points
    - This will select the corresponding columns in the Heat Map Detail panel
- In the Heat Map Detail Panel:
  - Click on a column dendrogram
    - This will highlight the corresponding points in the UMAP plot
  - Scroll over the Detail Map to zoom enough to see column labels at the bottom and click
   on a column label
    - This will highlight the corresponding point in the UMAP plot
- Click the
     <button class="viewer-close-icon">
        <svg style="width:16px;height:16px;vertical-align:middle;" viewBox="0 0 16 16">
          <use xlink:href="#icon-big-x"></use>
        </svg>
      </button> button to clear selections.

<button onclick="openIframeInNewTab('#myIframe')" style="margin-bottom:10px">Open NG-CHM in New Tab</button>


```{r, echo=FALSE, warning=FALSE, message=FALSE}
hm <- chmAddUMAP(hm, "column", umap_data)
invisible(chmExportToHTML(hm, "umapDemo.html", overwrite = TRUE))
library(htmltools)
htmltools::tags$iframe(src = "umapDemo.html", width = "100%", height = "800px", id = "myIframe", class = "ngchm-frame")
```

[Back to top](#)

## Further Reading

Additional examples and information are available in
[Introduction to Creating Single-Cell Next-Generation Clustered Heat Maps in R](https://www.ngchm.net/Downloads/how-to-create-single-cell-ngchm-in-r/tutorial-sc-deprez.pdf).

