<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Getting Started • NGCHMR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../lightswitch.js"></script><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Getting Started">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    <script> function openIframeInNewTab(cssSelector) { let iframe = document.querySelector(cssSelector); let url = iframe.src; window.open(url, '_blank'); } </script><nav class="navbar navbar-expand-lg fixed-top " aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">NGCHMR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.4</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/Installation.html">Installation</a></li>
<li class="nav-item"><a class="nav-link" href="../articles/getting-started.html">Getting Started</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-vignettes" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Vignettes</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-vignettes">
<li><a class="dropdown-item" href="../articles/getting-started.html">Getting Started</a></li>
    <li><a class="dropdown-item" href="../articles/color-maps.html">Color Maps</a></li>
    <li><a class="dropdown-item" href="../articles/data-layers.html">Data Layers</a></li>
    <li><a class="dropdown-item" href="../articles/ordering-and-clustering.html">Ordering and Clustering</a></li>
    <li><a class="dropdown-item" href="../articles/adding-linkouts.html">Adding Linkouts</a></li>
    <li><a class="dropdown-item" href="../articles/umap.html">UMAP</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Function Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-video-tutorials" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Video Tutorials</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-video-tutorials">
<li><a class="external-link dropdown-item" href="https://www.youtube.com/playlist?list=PLIBaINv-Qmd0v2qD4AMbGf85Xd-T_yzEh">Playlist: NG-CHMs in R/R-Studio</a></li>
    <li><a class="external-link dropdown-item" href="https://www.youtube.com/watch?v=O42w5P3A1_8&amp;list=PLIBaINv-Qmd0v2qD4AMbGf85Xd-T_yzEh">How to create NG-CHMs in R-Studio</a></li>
    <li><a class="external-link dropdown-item" href="https://www.youtube.com/watch?v=847breSor1I&amp;list=PLIBaINv-Qmd0v2qD4AMbGf85Xd-T_yzEh">How to use any ordering method with NG-CHMs</a></li>
    <li><a class="external-link dropdown-item" href="https://www.youtube.com/watch?v=K8PuHDQYk-g&amp;list=PLIBaINv-Qmd0v2qD4AMbGf85Xd-T_yzEh">How to create a single-cell NG-CHM in R</a></li>
    <li><a class="external-link dropdown-item" href="https://www.youtube.com/watch?v=iu0ptg9SBqM&amp;list=PLIBaINv-Qmd0v2qD4AMbGf85Xd-T_yzEh">Run NG-CHM RStudio container in Windows 10 Docker Desktop</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/MD-Anderson-Bioinformatics/NGCHM-R/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-lightswitch" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true" aria-label="Light switch"><span class="fa fa-sun"></span></button>
  <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="dropdown-lightswitch">
<li><button class="dropdown-item" data-bs-theme-value="light"><span class="fa fa-sun"></span> Light</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="dark"><span class="fa fa-moon"></span> Dark</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="auto"><span class="fa fa-adjust"></span> Auto</button></li>
  </ul>
</li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Getting Started</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/MD-Anderson-Bioinformatics/NGCHM-R/blob/v1.0.4/vignettes/getting-started.Rmd" class="external-link"><code>vignettes/getting-started.Rmd</code></a></small>
      <div class="d-none name"><code>getting-started.Rmd</code></div>
    </div>

    
    
<p>This vignette demonstrates how to:</p>
<ul>
<li><a href="#format-data-for-ng-chms">Format Data for NG-CHMs</a></li>
<li><a href="#create-an-ng-chm">Create an NG-CHM</a></li>
<li><a href="#add-a-covariate-bar">Add a covariate bar</a></li>
<li><a href="#export-to-html--ngchm-or-pdf">Export to HTML, .ngchm, or
PDF</a></li>
<li><a href="#embed-into-rmarkdown">Embed into Rmarkdown</a></li>
</ul>
<p>Complete code block examples are presented first in <a href="#r-code-examples">R Code Examples</a>, followed by the sections
detailing each step. The fully interactive NG-CHM is in the last
section, <a href="#resulting-ng-chm">Resulting NG-CHM</a>.</p>
<p>Packages used in these examples can be installed with:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"NGCHMDemoData"</span>, repos <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"https://md-anderson-bioinformatics.r-universe.dev"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"NGCHMSupportFiles"</span>, repos <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"https://md-anderson-bioinformatics.r-universe.dev"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"NGCHM"</span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="r-code-examples">R Code Examples<a class="anchor" aria-label="anchor" href="#r-code-examples"></a>
</h2>
<p>Below are two code examples. The first uses the <a href="https://md-anderson-bioinformatics.r-universe.dev/NGCHMDemoData" class="external-link">NGCHMDemoData</a>
to create an NG-CHM and export it to HTML and .ngchm files. The second
performs the same task with generic code containing placeholders for
user-specific data files, covariate column names, and other
variables.</p>
<details><summary>
Click to expand/collapse <strong>code example using
NGCHMDemoData</strong>
</summary><div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/MD-Anderson-Bioinformatics/NGCHMDemoData" class="external-link">NGCHMDemoData</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/MD-Anderson-Bioinformatics/NGCHMSupportFiles" class="external-link">NGCHMSupportFiles</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://md-anderson-bioinformatics.github.io/NGCHM-R/">NGCHM</a></span><span class="op">)</span></span>
<span><span class="va">matrix_data_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"TCGA.BRCA.Expression.csv"</span>, package <span class="op">=</span> <span class="st">"NGCHMDemoData"</span><span class="op">)</span></span>
<span><span class="va">matrix_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="va">matrix_data_file</span>, header <span class="op">=</span> <span class="cn">TRUE</span>, row.names <span class="op">=</span> <span class="fl">1</span>, check.names <span class="op">=</span> <span class="cn">FALSE</span>, stringsAsFactors <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">covariate_data_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"TCGA.BRCA.TP53Mutation.csv"</span>, package <span class="op">=</span> <span class="st">"NGCHMDemoData"</span><span class="op">)</span></span>
<span><span class="va">covariate_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="va">covariate_data_file</span>, row.names <span class="op">=</span> <span class="fl">1</span>, check.names <span class="op">=</span> <span class="cn">FALSE</span>, stringsAsFactors <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="co"># read.csv returns a data.frame</span></span>
<span><span class="va">covariate_vector</span> <span class="op">&lt;-</span> <span class="va">covariate_data</span><span class="op">[[</span><span class="st">"MutationState"</span><span class="op">]</span><span class="op">]</span> <span class="co"># create vector</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">covariate_vector</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">covariate_data</span><span class="op">)</span> <span class="co"># set the names</span></span>
<span><span class="va">hm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/chmNew.html">chmNew</a></span><span class="op">(</span><span class="st">"TCGA BRCA Expression"</span>, <span class="va">matrix_data</span><span class="op">)</span></span>
<span><span class="va">covariateBar</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/chmNewCovariate.html">chmNewCovariate</a></span><span class="op">(</span><span class="st">"TP53 Mutation"</span>, <span class="va">covariate_vector</span><span class="op">)</span></span>
<span><span class="va">hm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/chmAddCovariateBar-method.html">chmAddCovariateBar</a></span><span class="op">(</span><span class="va">hm</span>, <span class="st">"column"</span>, <span class="va">covariateBar</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/chmExportToHTML-method.html">chmExportToHTML</a></span><span class="op">(</span><span class="va">hm</span>, <span class="st">"tcga-brca.html"</span>, overwrite <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/chmExportToFile-method.html">chmExportToFile</a></span><span class="op">(</span><span class="va">hm</span>, <span class="st">"tcga-brca.ngchm"</span>, overwrite <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><a href="#">Back to top</a></p>
</details><details><summary>
Click to expand/collapse <strong>generic code example</strong> for .csv
input files
</summary><div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/MD-Anderson-Bioinformatics/NGCHMSupportFiles" class="external-link">NGCHMSupportFiles</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://md-anderson-bioinformatics.github.io/NGCHM-R/">NGCHM</a></span><span class="op">)</span></span>
<span><span class="co"># In the command below, replace "&lt;path to your matrix data file&gt;" with the path to your matrix data file.</span></span>
<span><span class="va">matrix_data_file</span> <span class="op">&lt;-</span> <span class="st">"&lt;path to your matrix data file&gt;"</span></span>
<span><span class="va">matrix_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="va">matrix_data_file</span>, header <span class="op">=</span> <span class="cn">TRUE</span>, row.names <span class="op">=</span> <span class="fl">1</span>, check.names <span class="op">=</span> <span class="cn">FALSE</span>, stringsAsFactors <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co"># In the command below, replace "&lt;path to your covariate data file&gt;" with the path to your covariate data file.</span></span>
<span><span class="va">covariate_data_file</span> <span class="op">&lt;-</span> <span class="st">"&lt;path to your covariate data file&gt;"</span></span>
<span><span class="va">covariate_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="va">covariate_data_file</span>, row.names <span class="op">=</span> <span class="fl">1</span>, stringsAsFactors <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="co"># read.csv returns a data.frame</span></span>
<span><span class="co"># In the command below, replace "&lt;covariate column name&gt;" with the name of the column in your covariate data file.</span></span>
<span><span class="va">covariate_vector</span> <span class="op">&lt;-</span> <span class="va">covariate_data</span><span class="op">[[</span><span class="st">"&lt;covariate column name&gt;"</span><span class="op">]</span><span class="op">]</span> <span class="co"># create vector</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">covariate_vector</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">covariate_data</span><span class="op">)</span> <span class="co"># set the names</span></span>
<span><span class="co"># In the command below, replace "&lt;name of your NG-CHM&gt;" with the desired name of your NG-CHM.</span></span>
<span><span class="va">hm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/chmNew.html">chmNew</a></span><span class="op">(</span><span class="st">"&lt;name of your NG-CHM&gt;"</span>, <span class="va">matrix_data</span><span class="op">)</span></span>
<span><span class="co"># In the command below, replace "&lt;name of your covariate&gt;" with the desired name of your covariate.</span></span>
<span><span class="va">covariateBar</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/chmNewCovariate.html">chmNewCovariate</a></span><span class="op">(</span><span class="st">"&lt;name of your covariate&gt;"</span>, <span class="va">covariate_vector</span><span class="op">)</span></span>
<span><span class="va">hm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/chmAddCovariateBar-method.html">chmAddCovariateBar</a></span><span class="op">(</span><span class="va">hm</span>, <span class="st">"column"</span>, <span class="va">covariateBar</span><span class="op">)</span></span>
<span><span class="co"># In the command below, replace "&lt;name of your HTML file&gt;" with the desired name of your HTML file.</span></span>
<span><span class="fu"><a href="../reference/chmExportToHTML-method.html">chmExportToHTML</a></span><span class="op">(</span><span class="va">hm</span>, <span class="st">"&lt;name of your HTML file&gt;"</span>, overwrite <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co"># In the command below, replace "&lt;name of your .ngchm file&gt;" with the desired name of your .ngchm file.</span></span>
<span><span class="co"># IMPORTANT: The filename must end with '.ngchm' in order to open in the NG-CHM Viewer</span></span>
<span><span class="fu"><a href="../reference/chmExportToFile-method.html">chmExportToFile</a></span><span class="op">(</span><span class="va">hm</span>, <span class="st">"&lt;name of your .ngchm file&gt;"</span>, overwrite <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><a href="#">Back to top</a></p>
</details>
</div>
<div class="section level2">
<h2 id="format-data-for-ng-chms">Format Data for NG-CHMs<a class="anchor" aria-label="anchor" href="#format-data-for-ng-chms"></a>
</h2>
<p>For use by the NGCHM package, data must be formatted such that:</p>
<ul>
<li>The heat map data is a matrix with rownames and colnames</li>
<li>The covariate data is a named vector.
<ul>
<li>At least one name in the row/column covariate data vector must be in
the row/column names of the heat map matrix.</li>
</ul>
</li>
</ul>
<p>This example uses the <a href="https://md-anderson-bioinformatics.r-universe.dev/NGCHMDemoData" class="external-link">NGCHMDemoData</a>
package, which can be installed from <a href="https://md-anderson-bioinformatics.r-universe.dev/packages" class="external-link">our
R-Universe repository</a>.</p>
<p>The NGCHMDemoData includes a matrix of gene expression data
containing 3437 genes (rows) and 200 samples (columns) of breast cancer
data from The Cancer Genome Atlas (TCGA). This will be converted into a
matrix for constructing the NG-CHM.</p>
<style>
table.expression-data {
  font-family: monospace;
  font-size: 12px;
  width: 90% !important;
  margin-left: auto !important;
  margin-right: auto !important;
}
table.expression-data>tbody>tr>td {
  padding: 5px;
  text-align: right;
}
table.expression-data>thead>tr>th {
  padding: 5px;
  text-align: right;
}
table.mutation-data {
  font-family: monospace;
  font-size: 12px;
  width: 50% !important;
  margin-left: auto !important;
  margin-right: auto !important;
}
table.mutation-data>tbody>tr>td {
  padding: 5px;
  text-align: center;
}
caption {
  text-align: center;
  font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
}
</style>
<table class="table expression-data">
<caption>
TCGA.BRCA.Expression.csv snippet
</caption>
<thead><tr class="header">
<th>
</th>
<th>
TCGA-AO-A0JJ-01A
</th>
<th>
TCGA-E9-A1R4-01A
</th>
<th>
TCGA-E9-A6HE-01A
</th>
<th>
TCGA-E2-A1L9-01A
</th>
<th>
...
</th>
</tr></thead>
<tbody>
<tr>
<td>
TSPAN6
</td>
<td>
11.838
</td>
<td>
9.483
</td>
<td>
11.157
</td>
<td>
11.536
</td>
<td>
...
</td>
</tr>
<tr>
<td>
CFH
</td>
<td>
12.033
</td>
<td>
10.835
</td>
<td>
10.824
</td>
<td>
12.878
</td>
<td>
...
</td>
</tr>
<tr>
<td>
ENPP4
</td>
<td>
11.372
</td>
<td>
10.749
</td>
<td>
9.721
</td>
<td>
11.008
</td>
<td>
...
</td>
</tr>
<tr>
<td>
SEMA3F
</td>
<td>
12.722
</td>
<td>
12.393
</td>
<td>
13.338
</td>
<td>
12.451
</td>
<td>
...
</td>
</tr>
<tr>
<td>
...
</td>
<td>
...
</td>
<td>
...
</td>
<td>
...
</td>
<td>
...
</td>
<td>
...
</td>
</tr>
</tbody>
</table>
<p><strong>IMPORTANT: In order to be used as the basis for an NG-CHM, a
matrix should have both rownames and colnames.</strong></p>
<p>The NGCHMDemoData also includes a vector of TP35 mutation status for
the TCGA samples in the matrix. This data will be used to construct a
covariate bar.</p>
<table class="table mutation-data">
<caption>
TCGA.BRCA.TP53Mutation.csv snippet
</caption>
<thead><tr class="header">
<th>
</th>
<th>
MutationState
</th>
</tr></thead>
<tbody>
<tr>
<td>
TCGA-AO-A0JJ-01A
</td>
<td>
WT
</td>
</tr>
<tr>
<td>
TCGA-E9-A1R4-01A
</td>
<td>
WT
</td>
</tr>
<tr>
<td>
TCGA-E9-A6HE-01A
</td>
<td>
WT
</td>
</tr>
<tr>
<td>
TCGA-E2-A1L9-01A
</td>
<td>
WT
</td>
</tr>
<tr>
<td>
TCGA-E9-A245-01A
</td>
<td>
WT
</td>
</tr>
<tr>
<td>
TCGA-AO-A0JG-01A
</td>
<td>
WT
</td>
</tr>
<tr>
<td>
TCGA-C8-A1HE-01A
</td>
<td>
WT
</td>
</tr>
<tr>
<td>
TCGA-C8-A12W-01A
</td>
<td>
MUT
</td>
</tr>
<tr>
<td>
...
</td>
<td>
...
</td>
</tr>
</tbody>
</table>
<p><strong>IMPORTANT: In order to be used as the basis for a row/column
covariate bar, the data should be formatted as a named vector with at
least one name in common with the row/column names of the
matrix.</strong></p>
<p>The NGCHMDemoData package includes these datasets as .csv files. In
the code example, the matrix data is read from the
TCGA.BRCA.Expression.csv file (in the NGCHMDemoData package), and
formatted as a matrix. The mutation data is read from the
TCGA.BRCA.TP53Mutation.csv file, and formatted as a named vector.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/MD-Anderson-Bioinformatics/NGCHMDemoData" class="external-link">NGCHMDemoData</a></span><span class="op">)</span></span>
<span><span class="co"># Read in the NGCHMDemoData expression data, and format as a matrix</span></span>
<span><span class="va">matrix_data_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"TCGA.BRCA.Expression.csv"</span>, package <span class="op">=</span> <span class="st">"NGCHMDemoData"</span><span class="op">)</span></span>
<span><span class="va">matrix_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="va">matrix_data_file</span>, header <span class="op">=</span> <span class="cn">TRUE</span>, row.names <span class="op">=</span> <span class="fl">1</span>, check.names <span class="op">=</span> <span class="cn">FALSE</span>, stringsAsFactors <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co"># Read in the NGCHMDemoData mutation data, and format as a named vector</span></span>
<span><span class="va">covariate_data_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"TCGA.BRCA.TP53Mutation.csv"</span>, package <span class="op">=</span> <span class="st">"NGCHMDemoData"</span><span class="op">)</span></span>
<span><span class="va">covariate_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="va">covariate_data_file</span>, row.names <span class="op">=</span> <span class="fl">1</span>, check.names <span class="op">=</span> <span class="cn">FALSE</span>, stringsAsFactors <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="co"># read.csv returns a data.frame</span></span>
<span><span class="va">covariate_vector</span> <span class="op">&lt;-</span> <span class="va">covariate_data</span><span class="op">[[</span><span class="st">"MutationState"</span><span class="op">]</span><span class="op">]</span> <span class="co"># create vector</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">covariate_vector</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">covariate_data</span><span class="op">)</span> <span class="co"># set the names</span></span></code></pre></div>
<p><a href="#">Back to top</a></p>
</div>
<div class="section level2">
<h2 id="create-an-ng-chm">Create an NG-CHM<a class="anchor" aria-label="anchor" href="#create-an-ng-chm"></a>
</h2>
<p>The <code><a href="../reference/chmNew.html">chmNew()</a></code> function creates the NG-CHM object. The
first argument is the name of the NG-CHM, and the second argument is the
matrix data.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://md-anderson-bioinformatics.github.io/NGCHM-R/">NGCHM</a></span><span class="op">)</span></span>
<span><span class="va">hm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/chmNew.html">chmNew</a></span><span class="op">(</span><span class="st">"TCGA BRCA Expression"</span>, <span class="va">matrix_data</span><span class="op">)</span></span></code></pre></div>
<p><a href="#">Back to top</a></p>
</div>
<div class="section level2">
<h2 id="add-a-covariate-bar">Add a Covariate Bar<a class="anchor" aria-label="anchor" href="#add-a-covariate-bar"></a>
</h2>
<p>The function <code><a href="../reference/chmNewCovariate.html">chmNewCovariate()</a></code> creates a covariate
object. The first argument is the name of the covariate, and the second
argument is the named vector of covariate data. The
<code><a href="../reference/chmAddCovariateBar-method.html">chmAddCovariateBar()</a></code> function adds the covariate to the
NG-CHM. The first argument is the NG-CHM, the second argument is the
desired map axis for the covariate bar (‘column’ or ‘row’), and the
third argument is the covariate object.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">covariateBar</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/chmNewCovariate.html">chmNewCovariate</a></span><span class="op">(</span><span class="st">"TP53 Mutation"</span>, <span class="va">covariate_vector</span><span class="op">)</span></span>
<span><span class="va">hm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/chmAddCovariateBar-method.html">chmAddCovariateBar</a></span><span class="op">(</span><span class="va">hm</span>, <span class="st">"column"</span>, <span class="va">covariateBar</span><span class="op">)</span></span></code></pre></div>
<p><a href="#">Back to top</a></p>
</div>
<div class="section level2">
<h2 id="export-to-html--ngchm-or-pdf">Export to HTML, .ngchm, or PDF<a class="anchor" aria-label="anchor" href="#export-to-html--ngchm-or-pdf"></a>
</h2>
<p>The NG-CHM can be exported to two different file types:</p>
<ol style="list-style-type: decimal">
<li>
<strong>HTML File</strong>: can be emailed to collaborators, opened
in a web browser, or embedded in an RMarkdown file.</li>
<li>
<strong>.ngchm File</strong>: can be opened in the <a href="https://www.ngchm.net/Downloads/ngChmApp.html" class="external-link">NGCHM
Viewer</a>.</li>
</ol>
<p>Both methods use files from the <a href="https://md-anderson-bioinformatics.r-universe.dev/NGCHMSupportFiles" class="external-link">NGCHMSupportFiles</a>
package. When loaded, NGCHMSupportFiles sets environment variables
pointing to these additional files.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/MD-Anderson-Bioinformatics/NGCHMSupportFiles" class="external-link">NGCHMSupportFiles</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="html-file">1. HTML File<a class="anchor" aria-label="anchor" href="#html-file"></a>
</h3>
<p>The function <code><a href="../reference/chmExportToHTML-method.html">chmExportToHTML()</a></code> exports the NG-CHM to an
HTML file. The first argument is the NG-CHM created above. The second
argument is the desired filename. The optional argument
<code>overwrite = TRUE</code> overwrites the file if it already
exists.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/chmExportToHTML-method.html">chmExportToHTML</a></span><span class="op">(</span><span class="va">hm</span>, <span class="st">"tcga-brca.html"</span>, overwrite <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>The file ‘tcga-brca.html’ can be shared with collaborators and opened
in a web browser.</p>
</div>
<div class="section level3">
<h3 id="ngchm-file">2. .ngchm File<a class="anchor" aria-label="anchor" href="#ngchm-file"></a>
</h3>
<p>Alternatively, <code><a href="../reference/chmExportToFile-method.html">chmExportToFile()</a></code> can be used to create a
.ngchm file.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/chmExportToFile-method.html">chmExportToFile</a></span><span class="op">(</span><span class="va">hm</span>, <span class="st">"tcga-brca.ngchm"</span>, overwrite <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>The file ‘tcga-brca.ngchm’ can be opened in the NG-CHM Viewer.
<strong>IMPORTANT: The filename must end with ‘.ngchm’ in order to open
in the NG-CHM Viewer</strong>.</p>
</div>
<div class="section level3">
<h3 id="pdf-file">3. PDF File<a class="anchor" aria-label="anchor" href="#pdf-file"></a>
</h3>
<p>To create a publication-quality PDF, there are several options:</p>
<!-- svg symbol symbol taken from NG-CHM viewer:
     https://github.com/MD-Anderson-Bioinformatics/NG-CHM/blob/main/NGCHM/WebContent/icons.svg?short_path=fffa0d9
-->
<svg style="display:none;"><symbol id="icon-menu-button-fill" xmlns="http://www.w3.org/2000/svg" fill="currentColor" class="bi bi-menu-button-fill" viewbox="0 0 16 16"><path d="M1.5 0A1.5 1.5 0 0 0 0 1.5v2A1.5 1.5 0 0 0 1.5 5h8A1.5 1.5 0 0 0 11 3.5v-2A1.5 1.5 0 0 0 9.5 0h-8zm5.927 2.427A.25.25 0 0 1 7.604 2h.792a.25.25 0 0 1 .177.427l-.396.396a.25.25 0 0 1-.354 0l-.396-.396zM0 8a2 2 0 0 1 2-2h12a2 2 0 0 1 2 2v5a2 2 0 0 1-2 2H2a2 2 0 0 1-2-2V8zm1 3v2a1 1 0 0 0 1 1h12a1 1 0 0 0 1-1v-2H1zm14-1V8a1 1 0 0 0-1-1H2a1 1 0 0 0-1 1v2h14zM2 8.5a.5.5 0 0 1 .5-.5h9a.5.5 0 0 1 0 1h-9a.5.5 0 0 1-.5-.5zm0 4a.5.5 0 0 1 .5-.5h6a.5.5 0 0 1 0 1h-6a.5.5 0 0 1-.5-.5z"></path></symbol></svg><ol style="list-style-type: decimal">
<li><p>Create an HTML file <a href="#html-file">as described above</a>.
Open the file in a web browser and create the view you want to save as a
PDF (e.g. the detail map zoomed to the desired location, any desired
rows/columns selected, covariate bars displayed, etc.). Then click the
<button class="fromViewer">
<svg style="width:1em;height:1em;vertical-align:middle;" viewbox="0 0 16 16"><use xlink:href="#icon-menu-button-fill"></use></svg></button> button
(top right corner) and select “Save Heat Map as PDF”.</p></li>
<li><p>Create a .ngchm file <a href="#ngchm-file">as described
above</a>. Open the .ngchm file in the <a href="https://tcga.ngchm.net/NGCHM/chm.html" class="external-link">NG-CHM Viewer</a> and
create the view you want to save as a PDF (e.g. the detail map zoomed to
the desired location, any desired rows/columns selected, covariate bars
displayed, etc.). Then click the <button class="fromViewer">
<svg style="width:1em;height:1em;vertical-align:middle;" viewbox="0 0 16 16"><use xlink:href="#icon-menu-button-fill"></use></svg></button> button
(top right corner) and select “Save Heat Map as PDF”.</p></li>
</ol>
<p><a href="#">Back to top</a></p>
</div>
</div>
<div class="section level2">
<h2 id="embed-into-rmarkdown">Embed into RMarkdown<a class="anchor" aria-label="anchor" href="#embed-into-rmarkdown"></a>
</h2>
<p>The HTML file can be embedded into RMarkdown as an iframe using
<code><a href="https://rstudio.github.io/htmltools/reference/builder.html" class="external-link">htmltools::tags</a></code>.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/rstudio/htmltools" class="external-link">"htmltools"</a></span><span class="op">)</span></span>
<span><span class="fu">htmltools</span><span class="fu">::</span><span class="va"><a href="https://rstudio.github.io/htmltools/reference/builder.html" class="external-link">tags</a></span><span class="op">$</span><span class="fu">iframe</span><span class="op">(</span>src <span class="op">=</span> <span class="st">"tcga-brca.html"</span>, width <span class="op">=</span> <span class="st">"100%"</span>, height <span class="op">=</span> <span class="fl">700</span><span class="op">)</span></span></code></pre></div>
<p><a href="#">Back to top</a></p>
</div>
<div class="section level2">
<h2 id="resulting-ng-chm">Resulting NG-CHM<a class="anchor" aria-label="anchor" href="#resulting-ng-chm"></a>
</h2>
<p>Below is the fully interactive NG-CHM created in this vignette.</p>
<iframe src="tcga-brca.html" width="100%" height="700"></iframe>
<p><a href="#">Bact to top</a></p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Bradley M Broom, Mary A Rohrdanz.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>



  <script> /* open external links in new tab */ (function() { for (const link of document.getElementsByTagName('a')) { if (link.classList.contains('external-link')) link.target = '_blank'; } })(); </script>
</body>
</html>
