<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>UMAP • NGCHMR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../lightswitch.js"></script><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="UMAP">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    <script> function openIframeInNewTab(cssSelector) { let iframe = document.querySelector(cssSelector); let url = iframe.src; window.open(url, '_blank'); } </script><nav class="navbar navbar-expand-lg fixed-top " aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">NGCHMR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.3</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/Installation.html">Installation</a></li>
<li class="nav-item"><a class="nav-link" href="../articles/getting-started.html">Getting Started</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-vignettes" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Vignettes</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-vignettes">
<li><a class="dropdown-item" href="../articles/getting-started.html">Getting Started</a></li>
    <li><a class="dropdown-item" href="../articles/color-maps.html">Color Maps</a></li>
    <li><a class="dropdown-item" href="../articles/data-layers.html">Data Layers</a></li>
    <li><a class="dropdown-item" href="../articles/ordering-and-clustering.html">Ordering and Clustering</a></li>
    <li><a class="dropdown-item" href="../articles/adding-linkouts.html">Adding Linkouts</a></li>
    <li><a class="dropdown-item" href="../articles/umap.html">UMAP</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Function Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-video-tutorials" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Video Tutorials</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-video-tutorials">
<li><a class="external-link dropdown-item" href="https://www.youtube.com/playlist?list=PLIBaINv-Qmd0v2qD4AMbGf85Xd-T_yzEh">Playlist: NG-CHMs in R/R-Studio</a></li>
    <li><a class="external-link dropdown-item" href="https://www.youtube.com/watch?v=O42w5P3A1_8&amp;list=PLIBaINv-Qmd0v2qD4AMbGf85Xd-T_yzEh">How to create NG-CHMs in R-Studio</a></li>
    <li><a class="external-link dropdown-item" href="https://www.youtube.com/watch?v=847breSor1I&amp;list=PLIBaINv-Qmd0v2qD4AMbGf85Xd-T_yzEh">How to use any ordering method with NG-CHMs</a></li>
    <li><a class="external-link dropdown-item" href="https://www.youtube.com/watch?v=K8PuHDQYk-g&amp;list=PLIBaINv-Qmd0v2qD4AMbGf85Xd-T_yzEh">How to create a single-cell NG-CHM in R</a></li>
    <li><a class="external-link dropdown-item" href="https://www.youtube.com/watch?v=iu0ptg9SBqM&amp;list=PLIBaINv-Qmd0v2qD4AMbGf85Xd-T_yzEh">Run NG-CHM RStudio container in Windows 10 Docker Desktop</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/MD-Anderson-Bioinformatics/NGCHM-R/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-lightswitch" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true" aria-label="Light switch"><span class="fa fa-sun"></span></button>
  <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="dropdown-lightswitch">
<li><button class="dropdown-item" data-bs-theme-value="light"><span class="fa fa-sun"></span> Light</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="dark"><span class="fa fa-moon"></span> Dark</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="auto"><span class="fa fa-adjust"></span> Auto</button></li>
  </ul>
</li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>UMAP</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/MD-Anderson-Bioinformatics/NGCHM-R/blob/main/vignettes/umap.Rmd" class="external-link"><code>vignettes/umap.Rmd</code></a></small>
      <div class="d-none name"><code>umap.Rmd</code></div>
    </div>

    
    
<p>This vignette demonstrates how to perform uniform manifold
approximation and projection (UMAP), add the resulting UMAP coordinates
to an NG-CHM, and explore the interactive features between
dimensionality reduction plots and the NG-CHM via the <a href="https://github.com/MD-Anderson-Bioinformatics/ScatterPlotPlugin" class="external-link">2D
Scatter Plot Plugin</a>. A similar analysis can be performed for use
with the <a href="https://github.com/MD-Anderson-Bioinformatics/ScatterPlotPlugin3D" class="external-link">3D
Scatter Plot Plugin</a>.</p>
<div class="section level2">
<h2 id="uniform-mainfold-approximation-and-projection-umap">Uniform Mainfold Approximation and Projection (UMAP)<a class="anchor" aria-label="anchor" href="#uniform-mainfold-approximation-and-projection-umap"></a>
</h2>
<p>The code block below reads in the <a href="https://md-anderson-bioinformatics.r-universe.dev/NGCHMDemoData" class="external-link">NGCHMDemoData</a>,
performs principal component analysis (PCA), and calculates UMAP
coordinates from the principal components. This vignette uses the <a href="https://cran.r-project.org/package=umap" class="external-link">umap R package</a> which
must be installed in order to run the code below. (See <a href="getting-started.html">Getting Started</a> for details on creating
NG-CHMs). A static plot of the UMAP coordinates, colored by TP53
mutation state is displayed in Figure 1.</p>
<p>PCA was performed first because it yielded better group separation
compared to performing UMAP on the raw data.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Read in NGCHMDemoData (as in the Getting Started vignette)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/MD-Anderson-Bioinformatics/NGCHMDemoData" class="external-link">NGCHMDemoData</a></span><span class="op">)</span></span>
<span><span class="va">matrix_data_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"TCGA.BRCA.Expression.csv"</span>, package <span class="op">=</span> <span class="st">"NGCHMDemoData"</span><span class="op">)</span></span>
<span><span class="va">matrix_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="va">matrix_data_file</span>, header <span class="op">=</span> <span class="cn">TRUE</span>, row.names <span class="op">=</span> <span class="fl">1</span>, check.names <span class="op">=</span> <span class="cn">FALSE</span>, stringsAsFactors <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">covariate_data_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"TCGA.BRCA.TP53Mutation.csv"</span>, package <span class="op">=</span> <span class="st">"NGCHMDemoData"</span><span class="op">)</span></span>
<span><span class="va">covariate_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="va">covariate_data_file</span>, row.names <span class="op">=</span> <span class="fl">1</span>, check.names <span class="op">=</span> <span class="cn">FALSE</span>, stringsAsFactors <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="co"># read.csv returns a data.frame</span></span>
<span><span class="va">covariate_vector</span> <span class="op">&lt;-</span> <span class="va">covariate_data</span><span class="op">[[</span><span class="st">"MutationState"</span><span class="op">]</span><span class="op">]</span> <span class="co"># create vector</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">covariate_vector</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">covariate_data</span><span class="op">)</span> <span class="co"># set the names</span></span>
<span><span class="co"># Calculate principal components</span></span>
<span><span class="va">pca_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/prcomp.html" class="external-link">prcomp</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">matrix_data</span><span class="op">)</span><span class="op">)</span>, scale <span class="op">=</span> <span class="cn">TRUE</span>, center <span class="op">=</span> <span class="cn">TRUE</span>, rank <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span><span class="co"># Calculate UMAP from principal components</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/tkonopka/umap" class="external-link">umap</a></span><span class="op">)</span></span>
<span><span class="va">config</span> <span class="op">&lt;-</span> <span class="fu">umap</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/umap/man/umap.defaults.html" class="external-link">umap.defaults</a></span></span>
<span><span class="va">config</span><span class="op">$</span><span class="va">n_neighbors</span> <span class="op">&lt;-</span> <span class="fl">15</span> <span class="co"># change default for better group separation</span></span>
<span><span class="va">config</span><span class="op">$</span><span class="va">random_state</span> <span class="op">&lt;-</span> <span class="fl">123</span> <span class="co"># set random state for reproducibility</span></span>
<span><span class="va">umap_data</span> <span class="op">&lt;-</span> <span class="fu">umap</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/umap/man/umap.html" class="external-link">umap</a></span><span class="op">(</span><span class="va">pca_data</span><span class="op">$</span><span class="va">x</span>, config <span class="op">=</span> <span class="va">config</span><span class="op">)</span></span></code></pre></div>
<details><summary>
Click to expand R code used to create static plot in Figure 1
</summary><div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Create static plot of UMAP coordinates colored by TP53 mutation state</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mar <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">4</span>, <span class="fl">4</span>, <span class="fl">4</span>, <span class="fl">8</span><span class="op">)</span> <span class="op">+</span> <span class="fl">0.1</span>, bg <span class="op">=</span> <span class="st">"white"</span>, mgp <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.5</span>, <span class="fl">1</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">xlim</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/range.html" class="external-link">range</a></span><span class="op">(</span><span class="va">umap_data</span><span class="op">$</span><span class="va">layout</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">ylim</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/range.html" class="external-link">range</a></span><span class="op">(</span><span class="va">umap_data</span><span class="op">$</span><span class="va">layout</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">xlim</span>, <span class="va">ylim</span>,</span>
<span>     xlab <span class="op">=</span> <span class="st">"UMAP 1"</span>,</span>
<span>     ylab <span class="op">=</span> <span class="st">"UMAP 2"</span>,</span>
<span>     main <span class="op">=</span> <span class="st">"UMAP"</span>,</span>
<span>     type <span class="op">=</span> <span class="st">"n"</span>, xaxt <span class="op">=</span> <span class="st">"n"</span>, yaxt <span class="op">=</span> <span class="st">"n"</span><span class="op">)</span></span>
<span><span class="va">labels_for_point_color</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="va">covariate_vector</span><span class="op">)</span> <span class="co"># "MUT' and "WT"</span></span>
<span><span class="va">colors</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"#f7ef81"</span>, <span class="st">"#ffc2e2"</span><span class="op">)</span> <span class="co"># plot only needs two colors: one for "MUT" and one for "WT"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/points.html" class="external-link">points</a></span><span class="op">(</span><span class="va">umap_data</span><span class="op">$</span><span class="va">layout</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span>, <span class="va">umap_data</span><span class="op">$</span><span class="va">layout</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span>,</span>
<span>       col <span class="op">=</span> <span class="va">colors</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/integer.html" class="external-link">as.integer</a></span><span class="op">(</span><span class="va">labels_for_point_color</span><span class="op">)</span><span class="op">]</span>,</span>
<span>       pch <span class="op">=</span> <span class="fl">19</span>, cex <span class="op">=</span> <span class="fl">1.5</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/legend.html" class="external-link">legend</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">xlim</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span> <span class="op">+</span> <span class="op">(</span><span class="va">xlim</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span> <span class="op">-</span> <span class="va">xlim</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span> <span class="op">*</span> <span class="fl">0.1</span>, <span class="co"># calculate x position for legend</span></span>
<span>       y <span class="op">=</span> <span class="va">ylim</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span>, <span class="co"># calculate y position for legend</span></span>
<span>       legend <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">labels_for_point_color</span><span class="op">)</span><span class="op">)</span>,</span>
<span>       col <span class="op">=</span> <span class="va">colors</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/integer.html" class="external-link">as.integer</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">labels_for_point_color</span><span class="op">)</span><span class="op">)</span><span class="op">]</span>,</span>
<span>       title <span class="op">=</span> <span class="st">"TP53 Mutation State"</span>,</span>
<span>       inset <span class="op">=</span> <span class="fl">0.03</span>, xpd <span class="op">=</span> <span class="cn">TRUE</span>, bty <span class="op">=</span> <span class="st">"n"</span>, pch <span class="op">=</span> <span class="fl">19</span>, cex <span class="op">=</span> <span class="fl">0.85</span><span class="op">)</span></span></code></pre></div>
</details><!-- If we evaluate the code in the details tag, the plot will be hidden when the
     deails tag is collapsed. So we have two copies of the plotting code: one in the details
     tag to let the user expand/collapse, and the second hidden but evalluated copy here to
     always display the plot.
--><div class="figure">
<img src="umap_files/figure-html/unnamed-chunk-4-1.png" alt="*Figure 1.* UMAP plot of TCGA BRCA data colored by TP53 mutation state" width="100%"><p class="caption">
<em>Figure 1.</em> UMAP plot of TCGA BRCA data colored by TP53 mutation
state
</p>
</div>
<p><a href="#">Back to top</a></p>
</div>
<div class="section level2">
<h2 id="add-umap-to-ng-chm">Add UMAP to NG-CHM<a class="anchor" aria-label="anchor" href="#add-umap-to-ng-chm"></a>
</h2>
<p>This section describes how to add the UMAP coordinates calculated <a href="#umap-coordinates">above</a> to the NG-CHM such that they can be
explored interactively via the 2D Scatter Plot plugin.</p>
<div class="section level3">
<h3 id="create-an-ng-chm">Create an NG-CHM<a class="anchor" aria-label="anchor" href="#create-an-ng-chm"></a>
</h3>
<p>This code block creates an NG-CHM from the data read in above and
creates a covariate bar for the TP53 mutation state. The colors of the
TP53 mutation state are chosen to match the colors in the UMAP plot
above.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://md-anderson-bioinformatics.github.io/NGCHM-R/">NGCHM</a></span><span class="op">)</span></span>
<span><span class="va">hm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/chmNew.html">chmNew</a></span><span class="op">(</span><span class="st">"TCGA BRCA Expression"</span>, <span class="va">matrix_data</span><span class="op">)</span></span>
<span><span class="va">colors</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"#f7ef81"</span>, <span class="st">"#ffc2e2"</span><span class="op">)</span> <span class="co"># same colors as in Figure 1</span></span>
<span><span class="va">mutationColorMap</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/chmNewColorMap.html">chmNewColorMap</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"MUT"</span>, <span class="st">"WT"</span><span class="op">)</span>, <span class="va">colors</span><span class="op">)</span></span>
<span><span class="va">covariateBar</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/chmNewCovariate.html">chmNewCovariate</a></span><span class="op">(</span><span class="st">"TP53 Mutation State"</span>, <span class="va">covariate_vector</span>, <span class="va">mutationColorMap</span><span class="op">)</span></span>
<span><span class="va">hm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/chmAddCovariateBar-method.html">chmAddCovariateBar</a></span><span class="op">(</span><span class="va">hm</span>, <span class="st">"column"</span>, <span class="va">covariateBar</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="add-umap-coordinates">Add UMAP Coordinates<a class="anchor" aria-label="anchor" href="#add-umap-coordinates"></a>
</h3>
<p>The UMAP coordinates are added to the NG-CHM via the convenience
function <code><a href="../reference/chmAddUMAP.html">chmAddUMAP()</a></code>. Similar functions exist for adding
PCA, TSNE, etc. See the “Add Scatter Plot Coordinates” section on the <a href="../reference/index.html#add-scatter-plot-coordinates">Function
Reference</a> page for more details. UMAP coordinates can also be added
in a fashion similar to the TP53 mutation state.</p>
<!-- Don't evaluate in order to avoid messages from chmExportToHTML (that can seemingly only
     be hidden with `invisible()`. The code block at the end of the vignette adds the umap
     covariate and generates the embedded NGCHM
-->
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/chmAddUMAP.html">chmAddUMAP</a></span><span class="op">(</span><span class="va">hm</span>, <span class="st">"column"</span>, <span class="va">umap_data</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/chmExportToHTML-method.html">chmExportToHTML</a></span><span class="op">(</span><span class="va">hm</span>, <span class="st">"umap.html"</span>, overwrite <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="co"># create HTML file of NG-CHM</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="interactive-umap-ng-chm">Interactive UMAP / NG-CHM<a class="anchor" aria-label="anchor" href="#interactive-umap-ng-chm"></a>
</h2>
<!-- svg symbols taken from NG-CHM viewer:
     https://github.com/MD-Anderson-Bioinformatics/NG-CHM/blob/main/NGCHM/WebContent/icons.svg?short_path=fffa0d9
-->
<svg style="display:none;"><symbol id="icon-four-panels" xmlns="http://www.w3.org/2000/svg" stroke="currentColor" viewbox="0 0 16 16"><rect x="0" y="0" width="8" height="8" fill="#e7fffc"></rect><rect x="8" y="0" width="8" height="8" fill="#ffe7e8"></rect><rect x="0" y="8" width="8" height="8" fill="#e7ffe9"></rect><rect x="8" y="8" width="8" height="8" fill="#fffbe7"></rect></symbol><symbol id="icon-gear" xmlns="http://www.w3.org/2000/svg" fill="currentColor" class="bi bi-gear-fill" viewbox="0 0 16 16"><path d="M9.405 1.05c-.413-1.4-2.397-1.4-2.81 0l-.1.34a1.464 1.464 0 0 1-2.105.872l-.31-.17c-1.283-.698-2.686.705-1.987 1.987l.169.311c.446.82.023 1.841-.872 2.105l-.34.1c-1.4.413-1.4 2.397 0 2.81l.34.1a1.464 1.464 0 0 1 .872 2.105l-.17.31c-.698 1.283.705 2.686 1.987 1.987l.311-.169a1.464 1.464 0 0 1 2.105.872l.1.34c.413 1.4 2.397 1.4 2.81 0l.1-.34a1.464 1.464 0 0 1 2.105-.872l.31.17c1.283.698 2.686-.705 1.987-1.987l-.169-.311a1.464 1.464 0 0 1 .872-2.105l.34-.1c1.4-.413 1.4-2.397 0-2.81l-.34-.1a1.464 1.464 0 0 1-.872-2.105l.17-.31c.698-1.283-.705-2.686-1.987-1.987l-.311.169a1.464 1.464 0 0 1-2.105-.872l-.1-.34zM8 10.93a2.929 2.929 0 1 1 0-5.86 2.929 2.929 0 0 1 0 5.858z"></path></symbol><symbol id="icon-big-x" xmlns="http://www.w3.org/2000/svg" fill="currentColor" viewbox="0 0 16 16"><path d="m0 0 1 0 7 7 7-7 1 0 0 1-7 7 7 7 0 1 -1 0-7-7-7 7-1 0 0-1 7-7-7-7z"></path></symbol></svg><p>This section describes how to use the <a href="https://bioinformatics.mdanderson.org/public-software/ngchm/" class="external-link">NG-CHM
Viewer</a> to explore the UMAP data interactively. NG-CHMs include
several plugins, among them a 2D Scatter Plot to allow for interactive
exploration of 2-dimensional data. Below are the steps to open this
plugin and use it to explore the UMAP coordinates.</p>
<p><em>It may be helpful to click the “Open NG-CHM in a New Tab” button
below and follow these steps in the larger space of a new tab.</em></p>
<ol style="list-style-type: decimal">
<li>In the “Heat Map Summary” panel, click the
<button class="fromViewer"> <svg class="fromViewer" viewbox="0 0 16 16"><use xlink:href="#icon-four-panels"></use></svg></button> button to
open the panel menu.</li>
<li>In the newly opened panel menu, under <span class="ngchm-panel-menu-headings">Panel Control</span>, select “<span class="ngchm-panel-menu">Add Panel Below</span>”.</li>
<li>In the newly created empty panel, click the
<button class="fromViewer">
<svg style="width:16px;height:16px;vertical-align:middle;" viewbox="0 0 16 16"><use xlink:href="#icon-four-panels"></use></svg></button> button, and
under <span class="ngchm-panel-menu-headings">Set content to</span>,
select “<span class="ngchm-panel-menu">2D ScatterPlot</span>”. This will
automatically open the <button class="fromViewer">
<svg class="fromViewer" viewbox="0 0 16 16"><use xlink:href="#icon-gear"></use></svg></button> menu for that
panel.</li>
<li>In the newly opened <button class="fromViewer">
<svg class="fromViewer" viewbox="0 0 16 16"><use xlink:href="#icon-gear"></use></svg></button> menu, keep the
default choices.</li>
<li>At the bottom of the open <button class="fromViewer">
<svg class="fromViewer" viewbox="0 0 16 16"><use xlink:href="#icon-gear"></use></svg></button> menu, click <span class="ngchm-gear-menu-button">APPLY</span> and then <span class="ngchm-gear-menu-button">CLOSE</span>.</li>
</ol>
<p>The UMAP scatter plot should be visible in the lower right panel of
the NG-CHM display.</p>
<p>Here are some suggestions for exploring the interactive features
between the UMAP plot and the NG-CHM:</p>
<ul>
<li>In the 2D Scatter Plot Panel:
<ul>
<li>Use the slider to increase the size of the points</li>
<li>Click the lasso button to enter Lasso/Select mode</li>
<li>Draw a lasso around a cluster of points
<ul>
<li>This will select the corresponding columns in the Heat Map Detail
panel</li>
</ul>
</li>
</ul>
</li>
<li>In the Heat Map Detail Panel:
<ul>
<li>Click on a column dendrogram
<ul>
<li>This will highlight the corresponding points in the UMAP plot</li>
</ul>
</li>
<li>Scroll over the Detail Map to zoom enough to see column labels at
the bottom and click on a column label
<ul>
<li>This will highlight the corresponding point in the UMAP plot</li>
</ul>
</li>
</ul>
</li>
<li>Click the <button class="viewer-close-icon">
<svg style="width:16px;height:16px;vertical-align:middle;" viewbox="0 0 16 16"><use xlink:href="#icon-big-x"></use></svg></button> button to clear
selections.</li>
</ul>
<button onclick="openIframeInNewTab('#myIframe')" style="margin-bottom:10px">
Open NG-CHM in New Tab
</button>
<iframe src="umapDemo.html" width="100%" height="800px" id="myIframe" class="ngchm-frame"></iframe>
<p><a href="#">Back to top</a></p>
</div>
<div class="section level2">
<h2 id="further-reading">Further Reading<a class="anchor" aria-label="anchor" href="#further-reading"></a>
</h2>
<p>Additional examples and information are available in <a href="https://www.ngchm.net/Downloads/how-to-create-single-cell-ngchm-in-r/tutorial-sc-deprez.pdf" class="external-link">Introduction
to Creating Single-Cell Next-Generation Clustered Heat Maps in
R</a>.</p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Bradley M Broom, Mary A Rohrdanz.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer>
</div>



  <script> /* open external links in new tab */ (function() { for (const link of document.getElementsByTagName('a')) { if (link.classList.contains('external-link')) link.target = '_blank'; } })(); </script>
</body>
</html>
